src/state/actions/dispatchArIOInteraction.ts

```diff
@@ -123,6 +126,92 @@ export default async function dispatchArIOInteraction({
           WRITE_OPTIONS,
         );
         break;
+      case ARNS_INTERACTION_TYPES.PRIMARY_NAME_REQUEST: {
+        dispatch({
+          type: 'setSigningMessage',
+          payload: 'Confirming Primary Name 1/2',
+        });
+        const existingPrimaryNameRequest = await arioContract
```

Comment:
There may be edge cases around this if a person has a previous request with a different name than the one they are currently trying to request, this may use the previous name.